che_docs.sh

```diff
@@ -0,0 +1,77 @@
+# available under the terms of the Eclipse Public License 2.0
+# which is available at https://www.eclipse.org/legal/epl-2.0/
+#
+# SPDX-License-Identifier: EPL-2.0
+#
+set -o pipefail
+
+CURRENT_VERSION=""
+RAW_CONTENT=""
+NEWLINE=$'\n'
+NEWLINEx2="$NEWLINE$NEWLINE"
+TABLE_HEADER="$NEWLINEx2,=== $NEWLINE Environment Variable Name,Default value, Description $NEWLINE"
+TABLE_FOOTER=",=== $NEWLINEx2"
+BUFF=""
+OUTPUT_PATH="src/main/pages/che-7/administration-guide/ref_configuring-system-variables.adoc"
+
+fetch_current_version() {
+  CURRENT_VERSION=$(grep -ri "<version>" pom.xml | tail -n 1 |sed -e "s/^[ \t]*<version>\([^<]*\)<.*$/\1/")
+  if [ $? -ne 0 ]; then
+    echo "Failure: Cannot read version from pom.xml" >&2
+    exit 1
+  fi
+  if [[ "$CURRENT_VERSION" == *-SNAPSHOT ]]; then
+     CURRENT_VERSION="master"
+  fi
+}
+
+
+fetch_conf_files_content() {
+  CHE_PROPERTIES_URL="https://raw.githubusercontent.com/eclipse/che/$CURRENT_VERSION/assembly/assembly-wsmaster-war/src/main/webapp/WEB-INF/classes/che/che.properties"
+  RAW_CONTENT=$(curl -sf "$CHE_PROPERTIES_URL")
+  if [ $? -ne 0 ]; then
+    echo "Failure: Cannot read che.properties from URL $CHE_PROPERTIES_URL" >&2
+    exit 1
+  fi
+  MULTIUSER_PROPERTIES_URL="https://raw.githubusercontent.com/eclipse/che/$CURRENT_VERSION/assembly/assembly-wsmaster-war/src/main/webapp/WEB-INF/classes/che/multiuser.properties"
+  RAW_CONTENT="$RAW_CONTENT $(curl -sf "$MULTIUSER_PROPERTIES_URL")"
+  if [ $? -ne 0 ]; then
+    echo "Failure: Cannot read multiuser.properties from URL $MULTIUSER_PROPERTIES_URL" >&2
+    exit 1
+  fi
+}
+
+parse_content() {
+  while read -r LINE
+  do
+    if [[ $LINE == '##'* ]]; then
+      if [[ ! -z $TOPIC ]]; then
+        BUFF="$BUFF$TABLE_FOOTER"                   # topic changes, closing the table
+      fi
+      TOPIC="${LINE//#}"                            # read topic stripping #-s
+      BUFF="$BUFF==${TOPIC} $TABLE_HEADER $NEWLINE" # new topic and table header
+    elif [[ $LINE == '#'* ]] && [[ ! -z $TOPIC ]]; then
+      TRIM_LINE=${LINE//#}                          # read description stripping #-s
+      DESCR_BUFF="$DESCR_BUFF${TRIM_LINE/ /}"       # collect all description lines into buffer
+    elif [[ -z $LINE ]] && [[ ! -z $TOPIC ]]; then
+      DESCR_BUFF=""                                 # empty line is a separator = cleanup description and property name + value
+      KEY=""
+      VALUE=""
+    elif [[ ! -z $TOPIC ]]; then
+      IFS=$'=' read KEY VALUE <<< $LINE             # property split into key and value
+      ENV=${KEY^^}                                  # capitalize property name
+      ENV="+${ENV//_/__}+"                          # replace single underscores with double
+      ENV=${ENV//./_}                               # replace dots with single underscore
+      BUFF="$BUFF $ENV,\"$VALUE\",\"${DESCR_BUFF//\"/\'}\" $NEWLINE"   # apply key value and description buffer
+    fi
+  done <<< $RAW_CONTENT
+  echo "$BUFF" >> $OUTPUT_PATH
```

Comment:
I'm not sure about just appending the content at the end of current file. What about having full table in extra file, generate it fully and import it from `static` adoc? Maybe warn if file already exist and add some force parameter to replace the file?

Reply:
```
Table from CSV data in file
|===
include::envvars.csv[]
|===
```
see https://asciidoctor.org/docs/asciidoc-syntax-quick-reference/

Reply:
I saw that. But we have abt. 20 tables, so it's inconvenient to generate 20 csv files and hardcode this names in template etc

Reply:
ok for me.

Reply:
I prefer multiple CSV to include, and no editable documentation content in the script.
So we keep the asciidoc file editable "normally" without hiding it in a script.
It makes easier to understand that the data in the CSV is generated.

Reply:
There is no editable content into script now. Bot for the multiple includes there is complexity since table quantity and  names are also dynamic so it's not know how to include them all in template. This is why i'm generating csv directly into template content.

Reply:
The script is altering  `src/main/pages/che-7/administration-guide/ref_configuring-system-variables.adoc`. We will have mixed content inside of it, and high risks to commit it by error. That is precisely what we can avoid with includes.

As an alternative, we could add a static include statement  and a fully generated file  with static name `src/main/pages/che-7/administration-guide/examples/system-variables.adoc`, added in .gitignore.

NB: titles shoud be like that

```
.My title
```

instead of the `== My title` form.

Reply:
Static include makes sense. Refactored.